app/dao/notifications_dao.py

```diff
@@ -104,13 +104,18 @@ def dao_get_last_date_template_was_used(template_id, service_id):
         if last_date_from_notifications:
             return last_date_from_notifications
 
-    last_date = (
-        db.session.query(functions.max(FactNotificationStatus.bst_date))
-        .filter(FactNotificationStatus.template_id == template_id, FactNotificationStatus.key_type != KEY_TYPE_TEST)
-        .scalar()
-    )
+    if db.session.query(
```

Comment:
The change makes sense. Am I right in thinking we will only hit this if statement if the notification has been deleted from the DB? That FactNotificationStatus is around longer.

Reply:
Yes, that's correct ☺️